### ISSUE TITLE
Add support for Cloud Management Service (cis) as a Destination in cloudsdk

### ISSUE THREAD

**Issue by shardulsonar** [SAP/cloud-sdk-js#3028]
**Is your feature request related to a problem? Please describe.**
* I am trying to use the cloud sdk for accessing the [Accounts Service API](https://api.sap.com/api/APIAccountsService/resource). 
* When I try to run the function 
```
        // Step 1 : Create Sub Account
        const response = await SubaccountOperationsApi.createSubaccount({
                displayName: "sub-account-name",
                region: "eu10",
                subdomain: "sub-account-name"
            },
            {
                subaccountAdmin: "xxxxx"
            }
        ).execute({
            destinationName: "cloud-management-service-central"
        })
```
* I am not completely sure what I must add to the VCAP services, but by checking the documentation I believe that there is no support for the credentials given by the cloud-management-service. 
* The binding looks something like this (The secret values are hidden) : 
```
{
  "endpoints": {
    "accounts_service_url": "https://accounts-service.cfapps.eu10.hana.ondemand.com",
    "cloud_automation_url": "https://cp-formations.cfapps.eu10.hana.ondemand.com",
    "entitlements_service_url": "https://entitlements-service.cfapps.eu10.hana.ondemand.com",
    "events_service_url": "https://events-service.cfapps.eu10.hana.ondemand.com",
    "external_provider_registry_url": "https://external-provider-registry.cfapps.eu10.hana.ondemand.com",
    "metadata_service_url": "https://metadata-service.cfapps.eu10.hana.ondemand.com",
    "order_processing_url": "https://order-processing.cfapps.eu10.hana.ondemand.com",
    "provisioning_service_url": "https://provisioning-service.cfapps.eu10.hana.ondemand.com",
    "saas_registry_service_url": "https://saas-manager.cfapps.eu10.hana.ondemand.com"
  },
  "grant_type": "client_credentials",
  "sap.cloud.service": "com.sap.core.commercial.service.central",
  "uaa": {
    "apiurl": "https://api.authentication.eu10.hana.ondemand.com",
    "clientid": "xxxx",
    "clientsecret": "xxxx",
    "credential-type": "binding-secret",
    "identityzone": "xxxx",
    "identityzoneid": "xxxx",
    "sburl": "https://internal-xsuaa.authentication.eu10.hana.ondemand.com",
    "subaccountid": "xxxx",
    "tenantid": "xxxx",
    "tenantmode": "shared",
    "uaadomain": "authentication.eu10.hana.ondemand.com",
    "url": "https://xxxx.authentication.eu10.hana.ondemand.com",
    "verificationkey": "-----BEGIN PUBLIC KEY-----\nXXXX\n-----END PUBLIC KEY-----",
    "xsappname": "xxxx",
    "xsmasterappname": "xxxx",
    "zoneid": "xxxx"
  }
}
```
* The error that I get 

```
[2022-11-03T21:17:25.895Z] WARN     (environment-accessor): No services of type destination found! This might cause errors in other parts of the application.
Error: Unable to get access token for "destination" service. No service instance of type "destination" found.
    at resolveService (/Users/I540081/dev/btp-api/node_modules/@sap-cloud-sdk/connectivity/src/scp-cf/environment-accessor.ts:201:13)
    at serviceToken (/Users/I540081/dev/btp-api/node_modules/@sap-cloud-sdk/connectivity/src/scp-cf/token-accessor.ts:41:27)
    at Function.getProviderServiceToken (/Users/I540081/dev/btp-api/node_modules/@sap-cloud-sdk/connectivity/src/scp-cf/destination/destination-from-service.ts:268:39)
    at Function.getDestinationFromDestinationService (/Users/I540081/dev/btp-api/node_modules/@sap-cloud-sdk/connectivity/src/scp-cf/destination/destination-from-service.ts:166:45)
```

**Describe the solution you'd like**

* The solution should be that I should be able to use the `execute` function with the right parameter without an error. 
* This means that we would need a transform function in [service-binding-to-destination.ts](https://github.com/SAP/cloud-sdk-js/blob/main/packages/connectivity/src/scp-cf/destination/service-binding-to-destination.ts)
* In this function I must provide only the URL and the remaining credentials can be automatically taken.   

**Describe alternatives you've considered**
* What I tried 
```
        const serviceBindingTransformFn = async (service: any) => ({
            name: service.name,
            label: service.label,
            credentials: {
                ...service.uaa,
            },
            url: service.endpoints.accounts_service_url || ""

        });
```
* I created a transform function, but somehow it does not seem to work, and I get an error that says `[AxiosError: Request failed with status code 404]`
* I have tried the entire flow using postman > to get the token and then fetch the actual URL, but somehow I am unable to make it work in cloud-sdk. 

**Impact / Priority**
* Impact is low. 

Affected development phase: e.g. Getting Started, Development, Release, Production
* Trying to automate BTP operations using cloud sdk. 

Timeline: Currently, we manually create sub-accounts, and we are already live. But it would be great if we can automate this :) 

**Additional context**

Not 100% sure if this is a feature request, or me not being able to use the API. 
Please let me know in the comments, Thanks 👍

**Comment by deekshas8** [comment]
Hi @shardulsonar,

As you already realized, for a custom service, you can use the `serviceBindingTransformFn` to transform the Service to a [Destination object](https://sap.github.io/cloud-sdk/api/2.10.0/interfaces/sap_cloud_sdk_connectivity.Destination). In addition to the `url`, you need a `client_credential` token to call the destination. Refer to the implementation [here](https://github.com/SAP/cloud-sdk-js/blob/main/packages/connectivity/src/scp-cf/destination/service-binding-to-destination.ts#L74) and use the `serviceToken()` to fetch the access token build your destination.

Regarding the 404 error, one probable cause might be that the service url called is wrong. To check that, can you please do the following:

As a first step, get the destination separately, using the destinationForServiceBinding function:

```
const serviceBindingTransformFn = async (service: Service) => {
  const transformedService = {
    ...service,
    credentials: { ...service.uaa }
  };
  const token = await serviceToken(transformedService);
 // build and return the destination 
};

const destination = await destinationForServiceBinding('cloud-management-service-central', {
  serviceBindingTransformFn ,
});
```

Then, execute the service using the `executeRaw(destination)` method. This gives you more details about the actual request. Please verify if the request url is correct. 

```
const response = await SubaccountOperationsApi.createSubaccount({
              displayName: "sub-account-name",
              region: "eu10",
              subdomain: "sub-account-name"
          },
          {
              subaccountAdmin: "xxxxx"
          }
      ).executeRaw(destination);
```

Regards,
Deeksha

**Comment by shardulsonar** [comment]
Hello @deekshas8, Thanks for the help with the hint I was able to make some progress. 
Here are a few points that I noticed: 
1. Regarding fetching the serviceToken : 
* > `const token = await serviceToken(transformedService)`

* This fails because it for some reason expects xsuaa credentials, which obviously are not created. This is because I am trying to create a new sub-account so I can never have one. 
* Full Error Message : 
```
Error: No binding to an XSUAA service instance found. Please make sure to bind an instance of the XSUAA service to your application.
    at selectXsuaaInstance (/Users/I540081/dev/btp-api/node_modules/@sap-cloud-sdk/core/src/connectivity/scp-cf/environment-accessor.ts:262:11)
    at getXsuaaServiceCredentials (/Users/I540081/dev/btp-api/node_modules/@sap-cloud-sdk/core/src/connectivity/scp-cf/environment-accessor.ts:211:7)
    at multiTenantXsuaaCredentials (/Users/I540081/dev/btp-api/node_modules/@sap-cloud-sdk/core/src/connectivity/scp-cf/token-accessor.ts:156:33)
    at /Users/I540081/dev/btp-api/node_modules/@sap-cloud-sdk/core/src/connectivity/scp-cf/token-accessor.ts:46:17
    at step (/Users/I540081/dev/btp-api/node_modules/@sap-cloud-sdk/core/dist/connectivity/scp-cf/token-accessor.js:44:23)
    at Object.next (/Users/I540081/dev/btp-api/node_modules/@sap-cloud-sdk/core/dist/connectivity/scp-cf/token-accessor.js:25:53)
    at /Users/I540081/dev/btp-api/node_modules/@sap-cloud-sdk/core/dist/connectivity/scp-cf/token-accessor.js:19:71
    at new Promise (<anonymous>)
    at __awaiter (/Users/I540081/dev/btp-api/node_modules/@sap-cloud-sdk/core/dist/connectivity/scp-cf/token-accessor.js:15:12)
    at serviceToken (/Users/I540081/dev/btp-api/node_modules/@sap-cloud-sdk/core/dist/connectivity/scp-cf/token-accessor.js:70:12)

```
* Not really sure why this occurs but I m guessing some mis-configuration in VCAP Services. 
* My VCAP Services : 
```
{
    "cis": [
        {
            "name": "cis-central",
            "endpoints": {
                "accounts_service_url": "https://accounts-service.cfapps.sap.hana.ondemand.com",
                "cloud_automation_url": "https://cp-formations.cfapps.sap.hana.ondemand.com",
                "entitlements_service_url": "https://entitlements-service.cfapps.sap.hana.ondemand.com",
                "events_service_url": "https://events-service.cfapps.sap.hana.ondemand.com",
                "external_provider_registry_url": "https://external-provider-registry.cfapps.sap.hana.ondemand.com",
                "metadata_service_url": "https://metadata-service.cfapps.sap.hana.ondemand.com",
                "order_processing_url": "https://order-processing.cfapps.sap.hana.ondemand.com",
                "provisioning_service_url": "https://provisioning-service.cfapps.sap.hana.ondemand.com",
                "saas_registry_service_url": "https://saas-manager.cfapps.sap.hana.ondemand.com"
            },
            "grant_type": "client_credentials",
            "sap.cloud.service": "com.sap.core.commercial.service.central",
            "uaa": {
                "clientid": "xxxxx",
                "clientsecret": "xxxx",
                "url": "https://xxxxx.authentication.sap.hana.ondemand.com",
                "identityzone": "xxxxx",
                "identityzoneid": "xxxx",
                "tenantid": "xxxx",
                "tenantmode": "shared",
                "sburl": "https://internal-xsuaa.authentication.sap.hana.ondemand.com",
                "apiurl": "https://api.authentication.sap.hana.ondemand.com",
                "verificationkey": "-----BEGIN PUBLIC KEY-----\nMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA53MxvBPOZ3dj7s6oMxs2\niSI9ZLJtfweqMC/lDUDdaRHKOXWMKXNaX4TZ4fWGL5xNpVkobTgkjMtiBTaWYGFt\nkW6UBj+AU7Xq64WaXAuH962VtmHmrq3/e19Et+TfgHd+VREPzFzbPfh86pYh1lkz\nwyc4R2pIFZwt1e6P58AeZEnwP0XlqAqmViL/2KIA+cHcHzPh6sjQMchQaonI45cc\nurkmeT0r7BVSBskc4cZjCJXQj+bgJ1qMTzj4VDMpeq3SBolzcpJfG7BQyUrbHUPn\nOkpzubvW70vnnisweWcE4LlajxcaUhP5IMI9ZEYl/7ixSpT55xZw1uj298/baT05\nawIDAQAB\n-----END PUBLIC KEY-----",
                "xsappname": "xxxxx",
                "subaccountid": "xxxx",
                "uaadomain": "authentication.sap.hana.ondemand.com",
                "zoneid": "xxxx",
                "credential-type": "binding-secret",
                "xsmasterappname": "cis-central!b13"
            }
        }
    ]
}
```
* Ideally I would expect the `serviceToken` function to give me a token with the credentials I have provided (`url`, `clientId`, `clientSecret`). 
* To get around this I implemented my own getToken function. which works correctly and provides me with a token. 

2. Regarding using the `destinationForServiceBinding` function, a slight modification worked. If I do not add the `ts-ignore`, ts compiler fails to recognize that both return the same value.  Error message is `TS2322: Type '(service: any) => Promise<Destination>' is not assignable to type '(serviceBinding: ServiceBinding) => Destination'.   Property 'url' is missing in type 'Promise<Destination>' but required in type 'Destination'.`

```
        const destination : Destination= await destinationForServiceBinding('cis-central', {
            // @ts-ignore
            transformationFn: serviceBindingTransformFn
        });
```

TL; DR; 
Your suggestion seemed to have worked but then I had to write the `serviceToken` function on my own.
